php/libraries/SinglePointLogin.class.inc

```diff
@@ -312,8 +261,28 @@ class SinglePointLogin
         if (password_verify($password, $oldhash)) {
             // Rehash according to latest standards if necessary.
             if (password_needs_rehash($oldhash, PASSWORD_DEFAULT)) {
-                $user = \User::factory($username);
-                $user->updatePassword($password);
+                try {
+                    $newHash = new \Password($password);
+                    \User::updatePassword(
+                        $newHash
+                    );
+                } catch (\InvalidArgumentException $e) {
+                    // Edge-case: if a user has a password based on old LORIS
+                    // password standards and it needs rehashing, it will
+                    // generate this exception when used above.
+                    // Rather than force a password change, we will manually
+                    // update the password here without using the Password class
```

Comment:
I think it would make more sense to just always do it the way that works in all situations, but I'm not going to block over it either.